be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -184,15 +184,17 @@ Status BetaRowsetReader::next_block(RowBlock** block) {
 Status BetaRowsetReader::next_block(vectorized::Block* block) {
     SCOPED_RAW_TIMER(&_stats->block_fetch_ns);
     if (config::enable_storage_vectorization && _context->is_vec) {
-        auto s = _iterator->next_batch(block);
-        if (!s.ok()) {
-            if (s.is_end_of_file()) {
-                return Status::OLAPInternalError(OLAP_ERR_DATA_EOF);
-            } else {
-                LOG(WARNING) << "failed to read next block: " << s.to_string();
-                return Status::OLAPInternalError(OLAP_ERR_ROWSET_READ_FAILED);
+        do {
+            auto s = _iterator->next_batch(block);
+            if (!s.ok()) {
+                if (s.is_end_of_file()) {
+                    return Status::OLAPInternalError(OLAP_ERR_DATA_EOF);
+                } else {
+                    LOG(WARNING) << "failed to read next block: " << s.to_string();
+                    return Status::OLAPInternalError(OLAP_ERR_ROWSET_READ_FAILED);
+                }
             }
-        }
+        } while (block->rows() == 0);
```

Comment:
After this PR, rows in block will never be 0 unless EOF. So maybe you also need to update codes in `_refresh_current_row`?